### ISSUE TITLE
'Aicc' in case of out_of_sample_size wrong?

### ISSUE THREAD

**Issue by mohitmunjal** [alkaline-ml/pmdarima#304]
```
def _aicc(model_results, nobs, add_constant):
    """Compute the corrected Akaike Information Criterion"""
    aic = model_results.aic

    # SARIMAX counts the constant in df_model if there was an intercept, so
    # only add one if no constant was included in the model
    df_model = model_results.df_model
    if add_constant:
        add_constant += 1  # add one for constant term
    return aic + 2. * df_model * (nobs / (nobs - df_model - 1.) - 1.)
```

```
 def update(self, y, exogenous=None, maxiter=None, **kwargs):
   .
   .
   . 
  # Get the model parameters, then we have to "fit" a new one. If you're
        # reading this source code, don't panic! We're not just fitting a new
        # arbitrary model. Statsmodels does not handle patching new samples in
        # very well, so we seed the new model with the existing parameters.
        params = model_res.params
        self._fit(y, exog, start_params=params, maxiter=maxiter, **kwargs)
```

When we use out_of_sample_size parameter in auto_arima, we fit the model using existing parameters as seed parameters and data as insample and outsample combined. I am guessing, retraining it will change the aic value too of the trained model( it should)

However, while calculating aicc, nobs is the number of observations in insample data. While the update routine was run on (insample + outsample) data. Shouldn't nobs be updated to account for that?

**Comment by mohitmunjal** [comment]
@tgsmith61591  any views on this?

**Comment by tgsmith61591** [comment]
> Shouldn't nobs be updated to account for that?

Is it not?

```python
>>> import pmdarima as pm
>>> from pmdarima.model_selection import train_test_split
>>> y = pm.datasets.load_wineind()
>>> train, test = train_test_split(y, train_size=125)
>>> mod = pm.ARIMA((1, 1, 2), (0, 1, 1, 12), suppress_warnings=True)
>>> mod.fit(train)
>>> mod.nobs_
125
>>> mod.update(test)
>>> mod.nobs_
176
```

Is that not what you would expect?

**Comment by mohitmunjal** [comment]
Yes. Got it. the last call to _fit in the update routine updates it. Thanks for clarifying. Closing it.
